### PR TITLE
Add delete api

### PR DESCRIPTION
**Pull request by ramsayleung**
1. add delete api for product admin.
2. add test function for product service.

### CODE REVIEW

File: src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java [hunk 1]
```diff
@@ -76,4 +78,20 @@ public ProductView createProduct(ProductDraft productDraft) {
...
+  /**
+   * Delete product by id.
+   *
+   * @param id id
+   * @param version version
+   */
+  public void deleteProductById(String id, Integer version) {
+    LOG.debug("Enter. ProductId: {}, version: {}.", id, version);
+    Product entity = productService.getProductById(id);
+    productService.validateVersion(version, entity);
```
Comment (davis999) [inline src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java]: there is a class named VersionValidator, use it is ok.

File: src/main/java/io/reactivesw/product/domain/service/ProductService.java [hunk 2]
```diff
@@ -157,4 +159,29 @@ private boolean isMatchingSku(Product product, String sku) {
...
     return result;
   }
+
+  /**
+   * Validate whether version matches or not.
+   *
+   * @param version version
+   * @param entity product
+   */
+  public void validateVersion(Integer version, Product entity) {
```
Comment (davis999) [inline src/main/java/io/reactivesw/product/domain/service/ProductService.java]: there is a class named VersionValidator, do not need this method.

File: src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java [hunk 3]
```diff
@@ -76,4 +78,20 @@ public ProductView createProduct(ProductDraft productDraft) {
...
+    productService.deleteProductById(id);
+    LOG.debug("Exit. Deleted productId: {}, version: {}.", id, version);
+    LOG.trace("Deleted product: {}.", entity);
```
Comment (davis999) [inline src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java]: may be log the product first and log exit will be better.

File: src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java (hunk 1 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java]: use VersionValidate class.

File: src/main/java/io/reactivesw/product/domain/service/ProductService.java (hunk 2 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/product/domain/service/ProductService.java]: use VersionValidate class.

File: src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java (hunk 3 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/product/application/admin/service/ProductApplication.java]: trace product first and then log exit.

**Pull request closed without merging**